addons/mail/models/mail_message.py

```diff
@@ -1171,11 +1171,19 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
-        updates = [[
-            (self._cr.dbname, 'res.partner', author.id),
-            {'type': 'message_notification_update', 'elements': self.env['mail.message'].concat(*author_messages)._message_notification_format()}
-        ] for author, author_messages in groupby(messages.sorted('author_id'), itemgetter('author_id'))]
-        self.env['bus.bus'].sendmany(updates)
+        data_dict = defaultdict(lambda: self.env['mail.message'])
+        for message in messages.sorted('author_id'):
+            partner = self.env.user.partner_id
+            author = message.author_id
+            data_dict[author.id] += message
+            if partner.id not in data_dict:
+                data_dict[partner.id] += message
```

Comment:
The condition should actually be on `message` not being in `data_dict[partner.id]`

But this can be replaced by using `|=` operator instead which does the condition internally.